src/app/home/HighlightCard.ts

```diff
@@ -25,3 +25,59 @@ export type HighlightCard = {
       href: URL,
   },
 }
+
+export type campaignFamilyName = 'christmasChallenge'|'summerGive'|'greenMatchFund'|'womenGirls'|'mentalHealthFund'|'artsforImpact';
+
+
+export type SfApiHighlightCard = Omit<HighlightCard, 'backgroundImageUrl'|'iconColor'|'button'> & {
+  campaignFamily: campaignFamilyName
+  button: {
+    text: string,
+    href: string
+  }
+};
+
+const replaceURLOrigin = (experienceUriPrefix: string, blogUriPrefix: string, donateGlobalUriPrefix: string, urlFromApi: string): URL => {
+
+  const href = urlFromApi
+    .replace('https://donate.biggive.org', donateGlobalUriPrefix)
+    .replace('https://biggive.org', blogUriPrefix)
+    .replace('https://community.biggive.org', experienceUriPrefix)
+  ;
+
+  return new URL(href);
+}
+
+export const SFAPIHighlightCardToHighlightCard = (experienceUriPrefix: string, blogUriPrefix: string, donateGlobalUriPrefix: string, sfApiHighlightCard: SfApiHighlightCard, ): HighlightCard => {
+
+  const campaignFamilyColours: Record<campaignFamilyName, brandColour> = {
+    christmasChallenge: 'brand-cc-red',
+    summerGive: 'brand-c4c-orange',
+    greenMatchFund: 'brand-gmf-green',
+    womenGirls: 'brand-wgmf-purple',
+    mentalHealthFund: 'brand-mhf-turquoise',
+    artsforImpact: 'brand-afa-pink',
+  };
+
+  const campaignFamilyBackgroundImages: Record<campaignFamilyName, URL> = {
+    christmasChallenge: new URL('/assets/images/card-background-cc-lights.jpg', donateGlobalUriPrefix),
```

Comment:
I would have liked to have either a test or a compile time check to make sure that these image files exist, and aren't typo'd or deleted when still required. But I couldn't find a way to do that, since the tests run in the browser and won't have access to the filesystem. I don't think it's essential.